common/src/main/java/org/keycloak/common/crypto/CryptoProvider.java

```diff
@@ -17,6 +15,14 @@ public interface CryptoProvider {
      * @return secureRandom implementation based on the available security algorithms according to environment (FIPS non-fips)
      */
     SecureRandom getSecureRandom() throws NoSuchAlgorithmException, NoSuchProviderException;
-
-    JWEAlgorithmProvider getAesKeyWrapAlgorithmProvider();
+    
+    /**
+     * Get some cryptographic utility implementation. Returned implementation can be dependent according to if we have
+     * non-fips bouncycastle or fips bouncycastle on the classpath. The implementation should be previously registered by {@link #registerUtility(String, Object)}
+     * 
+     * @param clazz Returned class.
+     * @param typeId Type of the utility, which we want to return 
+     * @return 
+     */
+    <T> T getCryptoUtility(Class<T> clazz, String typeId);
```

Comment:
I also agree this is a bit confusing.

Based on the discussions you had with @stianst and the constraints you pointed out doing this differently, I would just say we could have this method as `getAlgorithmProvider` because that is what we are actually interested here: obtaining different algorithm providers.

Wdyt?

Reply:
The former (typed) sounds better. But yeah, basically naming.